api/src/main/java/org/openmrs/module/aijar/dataintegrity/InvalidTBEncounters.java

```diff
@@ -0,0 +1,210 @@
+		String queryString = "SELECT o FROM Obs o WHERE o.voided = false AND o.encounter.patient.dead = 0 AND o.encounter.voided = 0 AND o.concept.uuid = :identifierConceptUuid AND o.valueText IN " 
+				+ " (SELECT ob.valueText FROM Obs ob WHERE ob.concept.uuid = :identifierConceptUuid AND ob.voided = 0 GROUP BY ob.concept.conceptId, ob.valueText HAVING COUNT(ob.valueText) > 1)"; 
+		Query query = getSession().createQuery(queryString);
+		query.setParameter("identifierConceptUuid", identifierConceptUuid);
+		
+		List<Obs> obsList = query.list();
+		Set<Patient> uniquePatientList = new HashSet<Patient>();
+		List<RuleResult<Patient>> ruleResults = new ArrayList<>();
+		for (Obs obs : obsList) {
+			Patient patient = obs.getEncounter().getPatient();
+			
+			RuleResult<Patient> ruleResult = new RuleResult<>();
+			ruleResult.setActionUrl("coreapps/patientdashboard/patientDashboard.page?patientId=" + patient.getUuid());
+			ruleResult.setNotes("Patient #" + getTbNumber(patient, obs.getEncounter()) + " with similar " + identifierTitle + " identifiers(" + obs.getValueText() + "): duplicated for a single patient,");
+			ruleResult.setEntity(patient);
+			
+			ruleResults.add(ruleResult);
+			uniquePatientList.add(patient);
+		}
+		log.info("There are " + uniquePatientList.size() + " Patients with similar " + identifierTitle + " identifiers: duplicated for a single patient,");
+		
+		return ruleResults;
+	}
+
+	/**
+	 * Patients with similar TB identifiers: duplicated for a single patient,
+	 * @return List<RuleResult<Patient>>
+	 */
+	public List<RuleResult<Patient>> singlePatientWithDuplicateTBNumberAcrossMultipleTreatmentPrograms(String identifierConceptUuid, String identifierTitle) {
+		log.info("Executing rule to find Patients with similar TB identifiers: duplicated for a single patient");
+		String queryString = "SELECT o FROM Obs o WHERE o.voided = false AND o.encounter.patient.dead = 0 AND o.encounter.voided = 0 AND o.concept.uuid = :identifierConceptUuid AND o.obsId IN " 
+				+ " (SELECT min(ob.obsId) FROM Obs ob WHERE ob.concept.uuid = :identifierConceptUuid AND ob.voided = 0 GROUP BY ob.concept.id, ob.valueText HAVING COUNT(ob.valueText) > 1)"; 
+		
+		Query query = getSession().createQuery(queryString);
+		query.setParameter("identifierConceptUuid", identifierConceptUuid);
+		
+		List<Obs> obsList = query.list();
+		Set<Patient> uniquePatientList = new HashSet<Patient>();
+		
+		List<RuleResult<Patient>> ruleResults = new ArrayList<>();
+		for (Obs obs : obsList) {
+			Patient patient = obs.getEncounter().getPatient();
+			RuleResult<Patient> ruleResult = new RuleResult<>();
+			ruleResult.setActionUrl("coreapps/patientdashboard/patientDashboard.page?patientId=" + patient.getUuid());
+			ruleResult.setNotes("Patient #" + getTbNumber(patient, obs.getEncounter()) + " with similar " + identifierTitle + " identifiers(" + obs.getValueText() + "): duplicated for a single patient,");
```

Comment:
Correct the label of this message: from "Patient # with similar District TB Number identifiers(D-1234): duplicated for a single patient,"

to "The District TB Number (D-1234) is used by another patient"